### PR TITLE
[WIP] Upgrade to prompt_toolkit 2.0

### PR DESCRIPTION
**Pull request by jonathanslenders**
These are the changes required for upgrading to prompt_toolkit 2.0 once it's released.

Do you need asynchronous autocompletion or not? By default, completion is synchronous, but we can make it async if completion takes too much time.

### CODE REVIEW

**Comment by jonathanslenders** [comment]
See also: https://github.com/jonathanslenders/python-prompt-toolkit/issues/568

**Comment by eliangcs** [comment]
@jonathanslenders thanks so much for the effort! We only need synchronous completion since the completion in http-prompt is simple and fast.

**Comment by AlphaKoder** [comment]
@jonathanslenders 

Could you please update your pull request to address the failing check?

I manually patched the cli.py and utils.py files (not requirements.txt - updated metadata.json instead) and your patch appears to work fine with prompt_toolkit 2.0.

Thanks,
-Shaun

**Comment by alansill** [comment]
Is there an update on this pull request?

**Comment by friederbluemle** [comment]
Hopefully this can be fixed/merged soon. Thanks!

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 click>=5.0
 httpie>=2.2.0
 parsimonious>=0.6.2
-prompt-toolkit>=1.0.0,<2.0.0
+prompt-toolkit>=2.0.0,<2.1.0
```
Comment (BoboTiG) [inline requirements.txt:4]:
```suggestion
prompt-toolkit>=2.0.0,<3.0.0
```

**Comment by BoboTiG** [comment]
Thanks a lot @jonathanslenders and sorry for the delay :)

**Pull request merged**